components/vc/front/src/cockpitLights.c

```diff
@@ -0,0 +1,109 @@
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+
+static struct
+{
+    bool imdState;
+    bool bmsState;
+}cockpitLight_data;
+/******************************************************************************
+*                       P U B L I C  F U N C T I O N S
+******************************************************************************/
+
+CAN_digitalStatus_E cockpitLight_imd_getStateCAN(void)
+{
+    return cockpitLight_data.imdState ? CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+}
+
+CAN_digitalStatus_E cockpitLight_bms_getStateCAN(void)
+{
+    return cockpitLight_data.bmsState ? CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+}
+
+static void cockpitLight_init(void)
+{
+    cockpitLight_data.imdState = LIGHT_ON;
+    drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+
+    cockpitLight_data.bmsState = LIGHT_ON;
+    drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_BMS_LIGHT_EN, DRV_IO_INACTIVE);
+}
+
+
+static void cockpitLight_periodic_10Hz(void)
+{
+    CAN_digitalStatus_E can_imd_status;
+    const bool IMD_LIGHT_valid = (CANRX_get_signal(VEH, BMSB_imdStatusMem, &can_imd_status) == CANRX_MESSAGE_VALID);    
+
+    if (IMD_LIGHT_valid == true)
+    {
+        switch (cockpitLight_data.imdState)
+        {
+            case CAN_DIGITALSTATUS_OFF:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+                break;
+            case CAN_DIGITALSTATUS_ON:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+            default:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+                break;
+        }
+    }
+    if (IMD_LIGHT_valid == false)
+    {
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+    }
+    CAN_digitalStatus_E can_bms_status;
+    const bool BMS_LIGHT_valid = (CANRX_get_signal(VEH, BMSB_bmsStatusMem, &can_bms_status) == CANRX_MESSAGE_VALID);
+    
+    if (BMS_LIGHT_valid == true)
+    {
+        switch (cockpitLight_data.bmsState) 
+        {
+            case CAN_DIGITALSTATUS_OFF:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+                break;
+            case CAN_DIGITALSTATUS_ON:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+            default:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+                break;
+        }
+    }
+    if (BMS_LIGHT_valid == false)
+    {
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+    }
```

Comment:
Where did the suggestion go? You should either answer it for why its not good or do that fix